src/Storages/tests/gtest_background_low_priority_processing_pool.cpp

```diff
@@ -0,0 +1,129 @@
+{
+    int policy;
+    sched_param param;
+    if (pthread_getschedparam(pthread_self(), &policy, &param))
+    {
+        std::lock_guard<std::mutex> guard(mut);
+        checks[i] = false;
+
+        return DB::BackgroundProcessingPoolTaskResult::ERROR;
+    }
+
+    if (policy != SCHED_IDLE)
+    {
+        std::lock_guard<std::mutex> guard(mut);
+        checks[i] = false;
+
+        return DB::BackgroundProcessingPoolTaskResult::ERROR;
+    }
+    return DB::BackgroundProcessingPoolTaskResult::SUCCESS;
+}
+
+DB::BackgroundProcessingPoolTaskResult odd(int *data)
+{
+    for (int i = 0; i < 10; ++i)
+    {
+        if (i % 2)
+        {
+            data[i] = i;
+        }
+    }
+    return DB::BackgroundProcessingPoolTaskResult::SUCCESS;
+}
+
+DB::BackgroundProcessingPoolTaskResult even(int *data)
+{
+    for (int i = 0; i < 10; ++i)
+    {
+        if (!(i % 2))
+        {
+            data[i] = i;
+        }
+    }
+    return DB::BackgroundProcessingPoolTaskResult::SUCCESS;
+}
+
+static const char * minimal_default_user_xml =
+"<yandex>"
+"    <profiles>"
+"        <default></default>"
+"    </profiles>"
+"    <users>"
+"        <default>"
+"            <password></password>"
+"            <networks>"
+"                <ip>::/0</ip>"
+"            </networks>"
+"            <profile>default</profile>"
+"            <quota>default</quota>"
+"        </default>"
+"    </users>"
+"    <quotas>"
+"        <default></default>"
+"    </quotas>"
+"</yandex>";
+
+static DB::ConfigurationPtr getConfigurationFromXMLString(const char * xml_data)
+{
+    std::stringstream ss{std::string{xml_data}};
+    Poco::XML::InputSource input_source{ss};
+    return {new Poco::Util::XMLConfiguration{&input_source}};
+}
+
+TEST(BackgroundLowPriorityProcessingPool, SimpleCase)
+{
+    using namespace DB;
+    const auto & context_holder = getContext();
+    Context ctx = context_holder.context;
+
+    ConfigurationPtr config = getConfigurationFromXMLString(minimal_default_user_xml);
+    ctx.setConfig(config);
+
+    int data[10];
+    BackgroundProcessingPool & pool = ctx.getBackgroundLowPriorityPool();
+
+    auto task_handle = pool.createTask([&data] { return odd(data); });
+    pool.startTask(task_handle, false);
+    auto even_handle = pool.createTask([&data] { return even(data); });
+    pool.startTask(even_handle, false);
+    std::this_thread::sleep_for(std::chrono::seconds(1));
+    pool.removeTask(task_handle);
+    pool.removeTask(even_handle);
+    for (int i = 0; i < 10; ++i)
+    {
+        ASSERT_EQ(i, data[i]) << "failed on " << i << " step\n";
+    }
+
+    std::mutex mut;
+    std::vector<bool> checks(16, true);
+    std::vector<BackgroundProcessingPool::TaskHandle> handles(16);
+    sched_param sched_param_{};
+    for (int i = 0; i < 16; ++i)
+    {
+        handles[i] = pool.createTask([&mut, &checks, i, &sched_param_] { pthread_setschedparam(pthread_self(), SCHED_IDLE, &sched_param_); return checkLow(checks, i, mut); });
```

Comment:
You should remove pthread_setschedparam from here - the whole point of this test is to check whether threads from the tested pool have their priority set to IDLE